controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/ScaleOperationTask.java

```diff
@@ -126,7 +127,19 @@ public ScaleOperationTask(final StreamMetadataTasks streamMetadataTasks,
                                 future = future.thenCompose(r -> streamMetadataStore.submitScale(scope, stream, scaleInput.getSegmentsToSeal(),
                                         new ArrayList<>(scaleInput.getNewRanges()), scaleInput.getScaleTime(), record, context, executor));
                             }
-                        } 
+                        } else if (!RecordHelper.verifyRecordMatchesInput(scaleInput.getSegmentsToSeal(), scaleInput.getNewRanges(), isManualScale, record.getObject())) {
+                            // ensure that we process the event only if the input matches the epoch transition record.
+                            // if its manual scale and the event doesnt match, it means the scale has already completed. 
+                            // for auto scale, it means another scale is on going and just throw scale conflict exception 
+                            // which will result in this event being postponed. 
+                            if (isManualScale) {
+                                log.info("Scale already completed.");
+                                return CompletableFuture.completedFuture(null);
+                            } else {
+                                log.info("Scale conflict.");
```

Comment:
Scale conflict already causes a WARN message to be thrown, Can we add more details to this log line.